src/Microsoft.AspNet.OData.Shared/GetNextPageHelper.cs

```diff
@@ -24,23 +24,36 @@ internal static Uri GetNextPageLink(Uri requestUri, IEnumerable<KeyValuePair<str
 
             int nextPageSkip = pageSize;
 
+            //If no value for skiptoken is provided; revert to using skip 
+            bool useSkipToken = !String.IsNullOrWhiteSpace(skipTokenValue);
+
             foreach (KeyValuePair<string, string> kvp in queryParameters)
             {
-                string key = kvp.Key;
+                string key = kvp.Key.ToLowerInvariant();
                 string value = kvp.Value;
+
+                if ((key == "$skip" && useSkipToken) || (key == "$skiptoken"))
```

Comment:
move $skiptoken to switch; and the skip to the switch statement as well.

Reply:
fixed.